### PR TITLE
resource/aws_security_group_rule: Properly handle updating description when protocol is -1/ALL

### LINKED ISSUE #1920: aws_security_group_rule: InvalidParameterValue: When protocol is ALL, you cannot specify from-port.
Hi there,
I found issue when I tried to update **description** of aws_security_group_rule.

### Terraform Version
- Terraform v0.10.7
- terraform-provider-aws_v1.1.0_x4

### Terraform Configuration Files
```hcl
resource "aws_security_group_rule" "sg-egress" {
  security_group_id = "${aws_security_group.sg.id}"
  type = "egress"
  from_port = 0
  to_port = 0
  protocol = -1
  cidr_blocks = ["0.0.0.0/0"]
  description = "all"
}

```

### Debug Output
```
$ terraform plan
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.network.aws_security_group_rule.sg-egress
      description: "" => "all"


Plan: 0 to add, 1 to change, 0 to destroy.
```

### Panic Output
```
$ terraform apply
module.network.aws_security_group_rule.sg-egress: Modifying... (ID: sgrule-xxxx)
  description: "" => "all"
Error applying plan:

1 error(s) occurred:

* module.network.aws_security_group_rule.sg-egress: 1 error(s) occurred:

* aws_security_group_rule.sg-egress: Error updating security group sg-xxxx rule description: InvalidParameterValue: When protocol is ALL, you cannot specify from-port.
	status code: 400, request id: xxxxx

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### References
- #1661 
- #1793

### PR DESCRIPTION
**Pull request by bflad**
Fixes #1920 

Previously:

```
--- FAIL: TestAccAWSSecurityGroupRule_Description_AllPorts (21.74s)
    testing.go:538: Step 2 error: Error applying: 1 error occurred:
        	* aws_security_group_rule.test: 1 error occurred:
        	* aws_security_group_rule.test: Error updating security group sg-0b6f5a54297754be3 rule description: InvalidParameterValue: When protocol is ALL, you cannot specify from-port.
```

Output from acceptance testing:

```
--- PASS: TestAccAWSSecurityGroupRule_Description_AllPorts (36.84s)
--- PASS: TestAccAWSSecurityGroupRule_Egress (21.08s)
--- PASS: TestAccAWSSecurityGroupRule_EgressDescription (21.13s)
--- PASS: TestAccAWSSecurityGroupRule_EgressDescription_updates (33.23s)
--- PASS: TestAccAWSSecurityGroupRule_ExpectInvalidCIDR (1.77s)
--- PASS: TestAccAWSSecurityGroupRule_ExpectInvalidTypeError (1.72s)
--- PASS: TestAccAWSSecurityGroupRule_Ingress_Classic (20.29s)
--- PASS: TestAccAWSSecurityGroupRule_Ingress_Ipv6 (42.93s)
--- PASS: TestAccAWSSecurityGroupRule_Ingress_Protocol (41.24s)
--- PASS: TestAccAWSSecurityGroupRule_Ingress_VPC (20.04s)
--- PASS: TestAccAWSSecurityGroupRule_IngressDescription (20.98s)
--- PASS: TestAccAWSSecurityGroupRule_IngressDescription_updates (35.99s)
--- PASS: TestAccAWSSecurityGroupRule_Issue5310 (20.12s)
--- PASS: TestAccAWSSecurityGroupRule_MultiDescription (82.07s)
--- PASS: TestAccAWSSecurityGroupRule_MultiIngress (22.57s)
--- PASS: TestAccAWSSecurityGroupRule_PartialMatching_basic (47.79s)
--- PASS: TestAccAWSSecurityGroupRule_PartialMatching_Source (44.23s)
--- PASS: TestAccAWSSecurityGroupRule_PrefixListEgress (51.97s)
--- PASS: TestAccAWSSecurityGroupRule_Race (274.72s)
--- PASS: TestAccAWSSecurityGroupRule_SelfReference (40.94s)
--- PASS: TestAccAWSSecurityGroupRule_SelfSource (41.11s)
```

### CODE REVIEW

File: aws/resource_aws_security_group_rule_test.go [hunk 1]
```diff
@@ -786,6 +786,57 @@ func TestAccAWSSecurityGroupRule_EgressDescription_updates(t *testing.T) {
...
+				ImportState:       true,
+				ImportStateIdFunc: testAccAWSSecurityGroupRuleImportStateIdFunc(resourceName),
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSSecurityGroupRuleConfigDescriptionAllPorts(rName, "description2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSecurityGroupRuleExists(securityGroupResourceName, &group),
+					testAccCheckAWSSecurityGroupRuleAttributes(resourceName, &group, &rule2, "ingress"),
+					resource.TestCheckResourceAttr(resourceName, "description", "description2"),
```
Comment (tombuildsstuff) [inline aws/resource_aws_security_group_rule_test.go]: do we also want to ensure the `from_port` / `to_port` values are set to empty values in the state on the way back?

Comment (bflad) [reply aws/resource_aws_security_group_rule_test.go]: Added these:

```go
resource.TestCheckResourceAttr(resourceName, "from_port", "0"),
resource.TestCheckResourceAttr(resourceName, "protocol", "-1"),
resource.TestCheckResourceAttr(resourceName, "to_port", "0"),
```

Still passes. 😄 

```
--- PASS: TestAccAWSSecurityGroupRule_Description_AllPorts (25.78s)
```

**Comment by bflad** [comment]
This has been released in [version 1.43.1 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1431-november-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by analytically** [comment]
This change keeps flagging my security groups as needing to be recreated:

```
-/+ module.nomad_sg.module.sg.aws_security_group_rule.ingress_with_self (new resource required)
      id:                       "sgrule-3623521784" => <computed> (forces new resource)
      description:              "Ingress Rule" => "Ingress Rule"
      from_port:                "0" => "-1" (forces new resource)
      protocol:                 "-1" => "-1"
      security_group_id:        "sg-052a15203f66daa75" => "sg-052a15203f66daa75"
      self:                     "true" => "true"
      source_security_group_id: "sg-052a15203f66daa75" => <computed>
      to_port:                  "0" => "-1" (forces new resource)
      type:                     "ingress" => "ingress"
```

At every apply.

**Comment by bflad** [comment]
Thanks for the heads up, @analytically. I have submitted #6423 to fix that regression, which was not previously covered by an acceptance test.

**Comment by analytically** [comment]
Awesome, thanks!!

**Pull request merged**